homeassistant/components/knx/storage/config_store.py

```diff
@@ -31,6 +31,20 @@ class KNXConfigStoreModel(TypedDict):
     entities: KNXEntityStoreModel
 
 
+class PlatformControllerBase(ABC):
+    """Entity platform controller base class."""
+
+    @abstractmethod
+    async def add_entity(self, unique_id: str, config: dict[str, Any]) -> None:
+        """Add a new entity."""
+
+    @abstractmethod
+    async def reload_with_new_config(
+        self, entity_entry: er.RegistryEntry, config: dict[str, Any]
+    ) -> None:
+        """Remove an existing entity and load it again with new config."""
```

Comment:
Removing the entity is a detail. The important thing is that the entity uses the new config, right? How exactly that is done is up to the platforms I think.